handler/capture/capture.go

```diff
@@ -0,0 +1,51 @@
+package capture
+
+import (
+	"sync"
+
+	"github.com/phemmer/sawmill/event"
+)
+
+// Handler captures events in a threadsafe slice.
+type Handler struct {
+	events []*event.Event
+	mutex  sync.Mutex
+}
+
+// NewHandler creates and returns a new Handler
+func NewHandler() *Handler {
+	return &Handler{
+		events: []*event.Event{},
+	}
+}
+
+// Event fills the sawmill.Handler interface
+func (handler *Handler) Event(logEvent *event.Event) error {
+	handler.mutex.Lock()
+	handler.events = append(handler.events, logEvent)
+	handler.mutex.Unlock()
+	return nil
+}
+
+// Last returns the last event captured
+func (handler *Handler) Last() *event.Event {
+	handler.mutex.Lock()
+	logEvent := handler.events[len(handler.events)-1]
+	handler.mutex.Unlock()
+	return logEvent
+}
+
+// Events returns the slice of captured events
+func (handler *Handler) Events() []*event.Event {
+	handler.mutex.Lock()
+	logEvents := handler.events
+	handler.mutex.Unlock()
+	return logEvents
```

Comment:
The returned value isn't safe. It's still a reference to the `handler.events`, which can be modified after it's been returned.